### ISSUE TITLE
[BERT][API] Custom classes for Pre-training, QA and classification/regression

### ISSUE THREAD

**Issue by eric-haibin-lin** [dmlc/gluon-nlp#496]
Now we have a generic `BERTModel` class which can be configured to perform encoding, pooling and decoding. It always returns the sequence encoding result, plus some additional output depending on the configuration. However, we may consider providing BERT classes for specific tasks (following https://github.com/huggingface/pytorch-pretrained-BERT#overview). Quote:

> - BertForPreTraining - BERT Transformer with masked language modeling head and next sentence prediction classifier on top (fully pre-trained)
> - BertForSequenceClassification - BERT Transformer with a sequence classification head on top (BERT Transformer is pre-trained, the sequence classification head is only initialized and has to be trained)
> - BertForQuestionAnswering - BERT Transformer with a token classification head on top (BERT Transformer is pre-trained, the token classification head is only initialized and has to be trained). (cc @fierceX)
> - BertForFeatureExtraction...

Why? This makes future optimizations easier when properly hybridized. If we know the task ahead of time, the model can be defined such that only useful results are returned in `forward()`. For example, BERTForPreTraining only need to return decoding result and next sentence prediction results, and does not need to return sequence encoding result. Doing so will lead to better memory utilization and faster speed.

@kenjewu

**Comment by bhsu22** [comment]
Hi, I realize this hasn't been worked on in over a year, but I think having the HuggingFace BERT specific classes would be helpful. At the very least, having these as examples might be useful for people jumping into gluonnlp. Is this issue still being worked on? fwiw, i'm trying to get something equivalent to the HuggingFace BertForSequenceClassification working.
